docs/tutorial/writing-tests/test-case.rst

```diff
@@ -0,0 +1,1148 @@
+    template-resource: network_iface_info
+    template-unit: job
+    id: network_available_{interface}
+    template-id: network_available_interface
+    template-filter:
+      network_iface_info.link_type == "ether" and network_iface_info.link_info_kind == ""
+    requires:
+      (network_iface_info.interface == "{interface}" and network_iface_info.operstate == "UP")
+    command:
+      echo Testing {interface}
+      ping -I {interface} 1.1.1.1 -c 1
+    _summary: Test that the internet is reachable via {interface}
+    flags: simple
+
+.. note::
+   For historical reasons the grammar of resource expressions is currently
+   broken. Even though they shouldn't be, parenthesis around this requires are
+   compulsory!
+
+Re-running the jobs we see the difference, now the jobs are there and skipped.
+The reason why they were skipped is clear from the output log (and the eventual
+submission).
+
+.. code-block:: none
+    :emphasize-lines: 6,7,12,13
+
+    (checkbox_venv) > checkbox-cli run com.canonical.certification::network_iface_info "com.canonical.certification::network_available_.*"
+    =========[ Running job 1 / 3. Estimated time left (at least): 0:00:00 ]=========
+    --------------[ Test that the internet is reachable via enp2s0f0 ]--------------
+    ID: com.canonical.certification::network_available_enp2s0f0
+    Category: com.canonical.plainbox::uncategorised
+    Job cannot be started because:
+     - resource expression '(network_iface_info.interface == "enp2s0f0" and network_iface_info.operstate == "UP")' evaluates to false
+    Outcome: job cannot be started
+    [...]
+    ==================================[ Results ]===================================
+     ☑ : Fetches information of all network intefaces
+     ☐ : Test that the internet is reachable via enp2s0f0
+     ☐ : Test that the internet is reachable via enp5s0
+     ☑ : Test that the internet is reachable via wlan0
+
+Let me conclude this section by highlighting this last point. See the
+difference between ``template-filter`` and ``requires``.
+
+- The resources filtered by the ``template-filter`` do not generate a test, we
+  do this when the generated test would not make sense (for example, connection
+  test for the loopback interface)
+- The resources that, when filtered by the ``resource`` expression is empty,
+  marks the job as skipped. We do this when the job makes sense (for example,
+  the interface exists) but the current situation makes it impossible for it
+  to pass for an external reason (for example, the ethernet port may work but
+  it is not currently plugged in)
+
+Dealing with complexity - Python
+================================
+
+The ``network_available`` test that we have created during this tutorial is
+very simple but, in the real world things are not as simple. For example,
+right now we are only pinging once from the test, if the ping goes through
+we call that a success, else a failure. This works in our simple scenario while
```

Comment:
```suggestion
the test is considered successful; otherwise, it's a failure. This works in our simple scenario while
```
a bit like a mixture of Python and English :)